app/services/draw_size_lock_toggler.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+#
+# Service object to toggle whether or not a group size is locked for a given
+# draw. Validates that the size is one for which there are currently available
+# suites.
+class DrawSizeLockToggler
+  # permit calling :toggle on the base class
+  def self.toggle(**params)
+    new(**params).toggle
+  end
+
+  # Initialize a new DrawSizeLockToggler
+  #
+  # @param draw [Draw] the draw in question
+  # @param size_str [String] the size param passed in the request
+  def initialize(draw:, size:)
+    @draw = draw
+    @errors = []
+    @size = parse_size(size)
+  end
+
+  def toggle
+    return error(errors.join(', ')) unless valid?
+    toggle_size_lock
+    return success if draw.save
+    error(draw.errors.full_messages.join(', '))
+  end
+
+  private
+
+  attr_reader :draw, :size
+  attr_accessor :errors
+
+  def parse_size(size)
+    errors << 'Invalid size' unless integer_string?(size)
+    size.to_i
+  end
+
+  def integer_string?(size)
+    /^\d+$/.match(size)
+  end
+
+  def valid?
+    errors.empty?
+  end
+
+  def toggle_size_lock
+    if draw.size_locked?(size)
+      draw.locked_sizes.delete_if { |s| s == size }
```

Comment:
Yup, I'm not sure that was 100% intentional but you're right, we couldn't do it the other way.